pennylane_qiskit/qiskit_device.py

```diff
@@ -458,6 +439,49 @@ def probabilities(self, wires=None):
         prob = np.apply_over_axes(np.sum, prob, inactive_wires).flatten()
         return OrderedDict(zip(basis_states, prob))
 
+    def eigvals(self, observable, wires, par):
+        """Determine the eigenvalues of observable(s).
+
+        Args:
+            observable (str, List[str]): the name of an observable,
+                or a list of observables representing a tensor product.
+            wires (List[int]): wires the observable(s) is measured on
+            par (List[Any]): parameters of the observable(s)
+
+        Returns:
+            array[float]: an array of size ``(len(wires),)`` containing the
+            eigenvalues of the observable
+        """
+        # observable should be Z^{\otimes n}
+        eigvals = z_eigs(len(wires))
+
+        if isinstance(observable, list):
+            # determine the eigenvalues
+            if "Hermitian" in observable:
+                # observable is of the form Z^{\otimes a}\otimes H \otimes Z^{\otimes b}
+                eigvals = np.array([1])
+
+                for k, g in itertools.groupby(
+                    zip(observable, wires, par), lambda x: x[0] == "Hermitian"
+                ):
+                    if k:
+                        p = list(g)[0][2]
+                        Hkey = tuple(p[0].flatten().tolist())
+                        eigvals = np.kron(eigvals, self._eigs[Hkey]["eigval"])
+                    else:
+                        n = len([w for sublist in list(zip(*g))[1] for w in sublist])
```

Comment:
This line seems to just be counting number of wires in a group of consecutive non=Hermitian operators. I wonder if there is an easier way to write this?

Reply:
Had a long think about it, but couldn't come up with anything better. Open to suggestions!